File: 1403. Minimum Subsequence in Non-Increasing Order/solution.py
class Solution(object):
  def minSubsequence(self, nums):
    """
    :type nums: List[int]
    :rtype: List[int]
    """
    nums.sort(reverse=True)
    total = 0
    for num in nums:
      total += num

    result = []
    resultTotal = 0
    for num in nums:
      result.append(num)
      resultTotal += num
      total -= num
      if resultTotal > total:
        break

    return result

File: 1403. Minimum Subsequence in Non-Increasing Order/solution.cpp
class Solution {
public:
  vector<int> minSubsequence(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    int total = 0;
    for (int index = 0; index < nums.size(); index++){
      int num = nums[index];
      total += num;
    }

    vector<int> result;
    int resultTotal = 0;
    for (int index = nums.size() - 1; index > -1; index--){
      int num = nums[index];
      result.push_back(num);
      resultTotal += num;
      total -= num;
      if (resultTotal > total){
        break;
      }
    }

    return result;
  }
};